stacks/spica/src/bucket/components/property-menu/property-menu.component.ts

```diff
@@ -0,0 +1,95 @@
+import {moveItemInArray} from "@angular/cdk/drag-drop";
+import {Component, Input, OnInit} from "@angular/core";
+import {MatDialog} from "@angular/material/dialog";
+import {MatSnackBar} from "@angular/material/snack-bar";
+import {Bucket} from "@spica-client/bucket/interfaces/bucket";
+import {AddFieldModalComponent} from "@spica-client/bucket/pages/add-field-modal/add-field-modal.component";
+import {BucketService} from "@spica-client/bucket/services/bucket.service";
+import {SnackbarError} from "@spica-client/core/layout/snackbar/interface";
+import {SnackbarComponent} from "@spica-client/core/layout/snackbar/snackbar.component";
+
+@Component({
+  selector: "app-property-menu",
+  templateUrl: "./property-menu.component.html"
+})
+export class PropertyMenuComponent implements OnInit {
+  @Input() schema: Bucket;
+  @Input() displayedProperties: string[];
+  @Input() property: any;
+  constructor(
+    private _snackBar: MatSnackBar,
+    private bs: BucketService,
+    private dialog: MatDialog
+  ) {}
+
+  ngOnInit(): void {}
+
+  deleteProperty(propertyKey: string) {
+    if (this.schema.primary === propertyKey) {
+      this._snackBar.openFromComponent(SnackbarComponent, {
+        data: {
+          message:
+            "In order to delete the primary field, you must first select another field as primary"
+        } as SnackbarError,
+        duration: 5000
+      });
+      return;
+    }
+
+    delete this.schema.properties[propertyKey];
+    if (this.schema.required && this.schema.required.includes(propertyKey)) {
+      this.schema.required.splice(this.schema.required.indexOf(propertyKey), 1);
+    }
+    this.saveSchema();
+  }
+  async editNewProperty(propertyKey: string = null) {
+    this.dialog.open(AddFieldModalComponent, {
+      width: "800px",
+      maxHeight: "90vh",
+      data: {
+        parentSchema: this.schema,
+        propertyKey: propertyKey
+      }
+    });
+  }
+
+  saveSchema() {
+    this.bs
+      .replaceOne(this.schema)
+      .toPromise()
+      .catch(err =>
+        this._snackBar.openFromComponent(SnackbarComponent, {
```

Comment:
Why did we need to open the snackbar manually? Error interceptor should handle these errors for all HTTP requests globally.